lib/CMakeLists.txt

```diff
@@ -32,3 +32,17 @@ set_target_properties(libspirv2clc PROPERTIES
 target_link_libraries(libspirv2clc SPIRV-Tools-opt)
 
 generate_export_header(libspirv2clc)
+
+add_library(SPIRV2CLC-Layer SHARED layer.cpp)
+
+target_include_directories(SPIRV2CLC-Layer
+  PRIVATE
+    ${OPENCL_HEADERS_DIR})
+
+target_compile_features(SPIRV2CLC-Layer PUBLIC cxx_std_17)
```

Comment:
String view was some cherry on top, but mainly because the layer singleton is a global inline variable defined in the header (second to last line). It's not critical, but inline globals were meant for just this. It can be changed to an extern variable.

Reply:
Ok. No need to change, it might be time to treat ourselves to C++17. If this triggers issues for someone we can always fall back to 14.